lab_1/main.py

```diff
@@ -36,7 +55,15 @@ def calculate_frequencies(tokens: list) -> dict:
     e.g. tokens = ['weather', 'sunny', 'man', 'happy']
     --> {'weather': 1, 'sunny': 1, 'man': 1, 'happy': 1}
     """
-    pass
+    freq_dict = {}
+    try:
+        for i in tokens:
+            if isinstance(i,str) and isinstance(tokens,list):
```

Comment:
Usually, there are checks for arguments and a function bode - on the same level.
Try to separate them and get rid of nesting.